confgenerator/logging.go

```diff
@@ -47,6 +48,20 @@ func setLogNameComponents(ctx context.Context, tag, logName, receiverType string
 	}.Components(ctx, tag, "setlogname")
 }
 
+func otelSetLogNameComponents(_ context.Context, logName, hostName string) []otel.Component {
+	return LoggingProcessorModifyFields{
+		Fields: map[string]*ModifyField{
+			// TODO: Prepend `receiver_id.` if it already exists, like the `fluent_forward` receiver?
+			"logName": {
+				DefaultValue: &logName,
+			},
+			`labels."compute.googleapis.com/resource_name"`: {
+				DefaultValue: &hostName,
+			},
+		},
```

Comment:
Don't we need to move `agent.googleapis.com/log_file_path` from `jsonPayload` to a label here? https://github.com/GoogleCloudPlatform/ops-agent/blob/master/confgenerator/logging.go#L40-L42

Reply:
No, that's only in the legacy config because fluent-bit populates the field in `jsonPayload`:

https://github.com/GoogleCloudPlatform/ops-agent/blob/7e24f4bca67a63cf43b6cf30bd326280263b0edf/confgenerator/logging_receivers.go#L117-L119

This is handled as part of the receiver pipeline in the otel version (though looking at the code it seems like it might be buggy - I should add some unit tests, since this isn't covered by transformation tests)

Reply:
It needs to end up being a label at any rate, so SGTM if you want to handle it in the receiver pipeline instead. (I assume you mean changing [this line](https://github.com/GoogleCloudPlatform/ops-agent/pull/1521/files#diff-92305d0589ed0ffcedd1368edad5c31bb62f7bd7098aa1150c02625391db628dR198) to write to an attribute instead of body.)

Reply:
I fixed the receiver, and added an integration test for this.